storage/rocksdb/rdb_datadic.h

```diff
@@ -112,6 +112,7 @@ const char RDB_UNPACK_DATA_TAG = 0x02;
 const size_t RDB_UNPACK_DATA_LEN_SIZE = sizeof(uint16_t);
 const size_t RDB_UNPACK_HEADER_SIZE =
     sizeof(RDB_UNPACK_DATA_TAG) + RDB_UNPACK_DATA_LEN_SIZE;
+const size_t RDB_COVERED_BITMAP_SIZE = sizeof(uint16_t);
```

Comment:
Should we have a third tag for values that have the covered_bitmap? While the index version of UPDATE3 should be sufficient information to indicate there is possible covered_bitmap for unpack data if there are any varchar columns, I think it would be safer if we used a new tag whenever we have to write out the covered_bitmap too.

Would this complicate the code unnecessarily?

So RDB_UNPACK_DATA_TAG = 2 would be used for all secondary index and primary indexes where there is not covered_bitmap. For any indexes that have a varchar key, then covered_bitmap can exist. The values for those keys should use RDB_UNPACK_COVERED_DATA_TAG = 3, which consists of 5 bytes: 1 for the tag, 2 for the size, 2 for the covered_bitmap. This way, we can consistently verify the DATA_TAG value matches our key columns types and catch problems early on.

Thoughts?